[SofaKernel] Minor code refactor in BaseData & new StringUtils functions.

damienmarchal opened the pull request:
Two change needed by the SofaPython3 plugin. 

- In BaseData Some method have two version, one without arguments the other with an ExecParm. In pybind11 we need to differentiate explicitely the two version making the code more complex. As I see no good reason why to keep the two version in Sofa I merged these two into one by using a default param = nullptr.  

- I have added a join/split function in the StringUtils class. 

- In SimpleAPI I found allowing BaseObjectDescription useful to create object and node. So I added new functions.
_____________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

damienmarchal committed:
[SofaKernel] CLEAN BaseData by factoring the duplicated methods.

guparan commented:
@damienmarchal Isn't this PR independent from #859? I can remove https://github.com/sofa-framework/sofa/pull/860/commits/ca88a838f21c73d4d4a81931f23f43b9ef379777 from it.

damienmarchal committed:
[SofaKernel] ADD in StringUtils function to split & join string into vector.

So it look taste like python instead of having to write complex stl algorithms.

damienmarchal committed:
[SofaKernel] ADD in SimpleAPI functions to create object and node from BaseObjectDescriptions.

guparan commented on SofaKernel/framework/sofa/helper/StringUtils.h:
```diff
@@ -23,29 +23,20 @@
 #include <vector>
 #include <sstream>
-
+#include "init.h"
```
Why this?

guparan commented:
[ci-build][with-scene-tests]

damienmarchal commented:
Can someone review this ?
